sdk/smx.ts

```diff
@@ -115,68 +121,146 @@ export class SMXStage {
     this.inputState$ = this.events.inputState$.map((value) => this.handleInputs(value));
   }
 
-  async init(): Promise<SMXConfig> {
+  async init(): Promise<SMXSensorTestData | undefined> {
     // Request the device information
-    this.updateDeviceInfo();
+    return this.updateDeviceInfo().then(() => {
+      // Requesting the correct config requires having the device info first
+      return this.updateConfig()?.then(() => {
+        // Requesting some initial test data requires we have the config first
+        // so that we know how to interpret the data
+        return this.updateTestData();
+      });
+    });
```

Comment:
Since you're already in an async function, this is the easier way to go about this:
```suggestion
    await this.updateDeviceInfo();
    // Requesting the correct config requires having the device info first
    await this.updateConfig();
    // Requesting some initial test data requires we have the config first
    // so that we know how to interpret the data
    return this.updateTestData();
```